discovery-provider/alembic/env.py

```diff
@@ -79,31 +79,42 @@ def run_migrations_online():
 
 @compiles(CreateIndex)
 @compiles(CreateTable)
+@compiles(AddColumn)
 def _add_if_not_exists(element, compiler, **kw):
     """Adds support for IF NOT EXISTS to CREATE TABLE and CREATE INDEX commands"""
     # Inspired by https://github.com/AudiusProject/audius-protocol/pull/2997
-    if isinstance(element, ddl.CreateIndex):
+    if isinstance(element, CreateIndex):
         output = compiler.visit_create_index(element, **kw)
-    elif isinstance(element, ddl.CreateTable):
+    elif isinstance(element, CreateTable):
         output = compiler.visit_create_table(element, **kw)
-    if element.element.info.get("if_not_exists"):
+    elif isinstance(element, AddColumn):
+        output = visit_add_column(element, compiler, **kw)
+    if not hasattr(element, "element") or element.element.info.get("if_not_exists"):
         output = re.sub(
-            "CREATE (TABLE|INDEX)", r"CREATE \g<1> IF NOT EXISTS", output, re.S
+            "(CREATE|ADD) (TABLE|INDEX|COLUMN)",
+            r"\g<1> \g<2> IF NOT EXISTS",
+            output,
+            re.S,
         )
     return output
 
 
 @compiles(DropIndex)
 @compiles(DropTable)
+@compiles(DropColumn)
 def _add_if_exists(element, compiler, **kw):
     """Adds support for IF EXISTS to DROP TABLE and DROP INDEX commands"""
     # Inspired by https://github.com/AudiusProject/audius-protocol/pull/2997
-    if isinstance(element, ddl.DropIndex):
+    if isinstance(element, DropIndex):
         output = compiler.visit_drop_index(element, **kw)
-    elif isinstance(element, ddl.DropTable):
+    elif isinstance(element, DropTable):
         output = compiler.visit_drop_table(element, **kw)
-    if element.element.info.get("if_exists"):
-        output = re.sub("DROP (TABLE|INDEX)", r"DROP \g<1> IF EXISTS", output, re.S)
+    elif isinstance(element, DropColumn):
+        output = visit_drop_column(element, compiler, **kw)
+    if not hasattr(element, "element") or element.element.info.get("if_exists"):
```

Comment:
Thinking of making add/drop columns idempotent by default, as alembic doesn't give us an element with kwargs to work with here. In which case, it might make sense to make create/drop index/table idempotent by default too? CC: @isaacsolo / @jowlee ?

Reply:
yeah that makes sense to me! this is pretty cool. are you suggesting we use ORM for DDL stuff instead of raw SQL?

Reply:
Ultimate goal is to just write a model then use `alembic revision --autogenerate` and have migrations made for us!